contracts/lens/ZapAnalyzer.sol

```diff
@@ -250,23 +252,20 @@ contract ZapAnalyzer is IZapAnalyzer {
     /// @notice Returns value based on other token
     /// @param fullPath swap path
     /// @return weightedPrice value of last token of path based on first
-    function getWeightedPrice(
-        SwapPath[] memory fullPath,
-        uint256 amount,
-        bool exact
-    ) internal view returns (uint256 weightedPrice) {
+    function getWeightedPrice(SwapPath[] memory fullPath) internal view returns (uint256 weightedPrice) {
         weightedPrice = 1e18;
         for (uint256 i = 0; i < fullPath.length; i++) {
             SwapPath memory path = fullPath[i];
             if (path.swapType == SwapType.V2) {
+                //divide by 1/6 to decrease slippage impact. Noted that it only works for tokens with 6 or more decimals
+                uint256 amount = 10**IERC20Metadata(path.path[0]).decimals() / 1e6;
```

Comment:
Lets take btc-quick as example. Btc at $30000 and quick at $0.05. lets say there's $60k worth in pool. $30k btc (1) and $30k quick (600.000). First it tried estimating swapping numbers with 1 full btc with high slippage. So it estimates like 200.000 quick output instead of the 600.000.
So maybe you add just 0.001 btc and thus within zap 2000 quick to the lp.
With estimation of 1e12 btc the slippage is less and you add 0.001 btc with 5900 quick which is way closer the actual ratio. Just random guessing example numbers but hope you get the point. And for gamma it just failed before because it needs be inside a specific small ratio

Reply:
Ok, this is helpful thanks. 

I added something more dynamic like this:
```
                uint256 precision = 1e6; // Use a precision of 1e6 for tokens with more than 12 decimals
                if (fullTokenAmount <= 1e12) {
                    precision = 1e4; // Use a precision of 1e4 for tokens with more than 6 decimals
                } else if (fullTokenAmount <= 1e6) {
                    precision = 1e2; // Use a precision of 1e2 for tokens with less than or equal to 6 decimals
                }
```